src/data_processor.py

```diff
@@ -1,6 +1,160 @@
-"""Process data retrieved by data_miner.py."""
-# import data_collection
+"""A file for calculating how well a team works together using the interquartile range.
 
+and calculate the individual contribution as a percentage of overall contribution
+"""
 
-def process_data():
-    """Process the calculated data."""
+# This import fixes a linting error with old division.
+from __future__ import division
+from collections import defaultdict
+import numpy as np
+import data_collection
+
+
+def iterate_nested_dictionary(dictionary):
+    """Iterate through the nested dictionary and create new dictionary."""
+    # iterate through nested dictionary and add to new dictionary
+    category_scores = {}
+    amount = 0
+    for user in dictionary.keys():
+        for metric in dictionary[user]:
+            # disregard any instances of strings in dictionary
+            if not isinstance(dictionary[user][metric], str):
+                # if list, add the length of the list to new dictionary
+                if isinstance(dictionary[user][metric], list):
+                    amount = len(dictionary[user][metric])
```

Comment:
If this is intended to calculate totals from all users, should this be `+=` instead of just `=` I'm not sure I understood it correctly.

Reply:
@noorbuchi This is an assignment, only if the metric has not yet been added to the dictionary, `.append` is used to append the dictionary after it is already created. This is the same for the below comment.